src/main/java/echoserver/ClientHandler.java

```diff
@@ -0,0 +1,51 @@
+package echoserver;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.net.Socket;
+
+class ClientHandler implements Runnable {
+    private final Socket socket;
+    private BufferedReader input;
+    private PrintWriter output;
+
+    ClientHandler(Socket socket) {
+        this.socket = socket;
+    }
+
+    @Override
+    public void run() {
+        setInputAndOutput();
+        sendAndReceiveMessages();
+        closeConnection();
+    }
+
+    private void setInputAndOutput() {
+        try {
+            var inputStreamReader = new InputStreamReader(socket.getInputStream());
+            input = new BufferedReader(inputStreamReader);
+            output = new PrintWriter(socket.getOutputStream(), true);
+        } catch (IOException e) {
```

Comment:
Just curious - did you look up how to test catching exceptions? Would you find such a test useful?